### PR TITLE
Feature/telegram notifications

### PR DESCRIPTION
**Pull request by onedr0p**
This is a feature request filled for #51 @drzoidberg33 If you can, please merge in with current dev branch because that has the on_created notifications. Thanks!

FYI, I choose to use the [leandrotoledo/python-telegram-bot](https://github.com/leandrotoledo/python-telegram-bot/blob/master/examples/roboed.py) library in case of future updates to the Telegram API. Let me know if this is a problem.

### CODE REVIEW

**Comment by drzoidberg33** [comment]
Thanks for the PR. I'll try and review this a bit later today.

**Comment by onedr0p** [comment]
Thanks for your continued support on this project, I'll see what I can do to help.

**Comment by drzoidberg33** [comment]
Sorry, I made comments on your fork by accident, but check those out. Otherwise it looks good.

I'm also wondering if we need such a massive Telegram library to just send messages, is there anything more lightweight that we can use?

**Comment by drzoidberg33** [comment]
I'm wondering now if there is any reason to even include that library. We're using one endpoint, which is a simple HTTP request.

We'll also be able to catch and log our errors more cleanly.

Anything I'm missing?

**Comment by onedr0p** [comment]
Telegram seems to be pretty active in their development, I included the library in case they change their api. That and it was easier to put together. I can look at adding support without the lib if your interested, or look for a more lightweight one.

**Comment by onedr0p** [comment]
I took your advice and made it work without the lib, it just uses a simple HTTP request.

I tested it for a bit and didn't see any problems.

**Comment by drzoidberg33** [comment]
Sorry, been super busy and haven't had time to review the new changes yet. Will take a look tonight hopefully.

File: plexpy/notifiers.py [hunk 1]
```diff
@@ -1417,3 +1437,68 @@ def return_config_options(self):
...
+        return cherrypy.config['config'].get('Telegram', options)
+
+    def notify(self, message, event):
+        if not message or not event:
+            return
+
+        http_handler = HTTPSConnection("api.telegram.org")
+
+        data = {'chat_id': self.chat_id,
+                'text': message.encode("utf-8")}
```
Comment (drzoidberg33) [inline plexpy/notifiers.py]: We're dropping the message subject here, can't we do something like with the Twitter notifications and format it like:

Subject: Message

Comment (onedr0p) [inline plexpy/notifiers.py]: I think I understand.

``` python
elif config_id == 13:
          telegramClient = TELEGRAM()
          telegramClient.notify(message=body, event=subject)
```

I pass the subject as event, so is this what you mean?

``` python
data = {'chat_id': self.chat_id,
    'text': event + ': ' + message.encode("utf-8")}
```

Comment (onedr0p) [inline plexpy/notifiers.py]: Updated the PR to include this.

File: plexpy/notifiers.py [hunk 2]
```diff
@@ -1417,3 +1437,68 @@ def return_config_options(self):
...
+        return cherrypy.config['config'].get('Telegram', options)
+
+    def notify(self, message, event):
+        if not message or not event:
+            return
+
+        http_handler = HTTPSConnection("api.telegram.org")
+
+        data = {'chat_id': self.chat_id,
+                'text': event + ': ' + message.encode("utf-8")}
```
Comment (drzoidberg33) [inline plexpy/notifiers.py]: `event` should be encoded too.

Comment (onedr0p) [inline plexpy/notifiers.py]: updated in PR

**Comment by drzoidberg33** [comment]
Thanks! Merging.

**Comment by onedr0p** [comment]
sweet, looking forward to the new release!

**Pull request merged**